### PR TITLE
Feat/app/new error handling

### PR DESCRIPTION
**Pull request by tomasdiaztoro**
Add new exceptions handling, including error codes.
Status is no included in error responses.

Related backed PR: https://github.com/cornershop/cornershop-backend/pull/17034

### CODE REVIEW

File: integration/app.py [hunk 1]
```diff
@@ -80,88 +89,85 @@ def create_check():
...
+            logger.info("BGC adapter service unavailable exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except TimeoutAPIException as e:
+            logger.info("BGC adapter timeout exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except UnauthorizedAPIException as e:
+            logger.info("BGC adapter unauthorized exception", extra=get_logger_data(check_data, e.error_message))
+            return get_error_response(e, 400)
+        except UnprocessableEntityAPIException as e:
+            logger.info("BGC adapter unprocessable entity exception", extra=get_logger_data(check_data, e.error_message))
```
Comment (richardbm) [inline integration/app.py:160]: We just need to catch `GenericAPIException`. all exceptions raised that extend that one will be caught. The logs could get the message from the exception.  maybe `logger.info("BGC request error {e.error_message}"`

we could also have a decorator for the view that caught `GenericAPIException` and log and return the error message. What do you think @tomasdiaztoro @jeffersonlizar ?

Comment (tomasdiaztoro) [reply integration/app.py:160]: agree with the first, the decorator seems a bit unneeded

Comment (tomasdiaztoro) [reply integration/app.py:160]: but we said that the Generic exception must reach sentry 👁️

Comment (richardbm) [reply integration/app.py]: We need to raise an exception for not handled responses. we were using `GenericAPIException`. but we could also define another one that does not extend from GenericAPIException.

Comment (richardbm) [reply integration/app.py]: the use of the decorator for the exception handling is optional. I don't have a strong opinion thee

File: integration/app.py [hunk 2]
```diff
@@ -80,88 +81,49 @@ def create_check():
...
+            return get_error_response(e, 400)
 
     @app.route("/get_check", methods=["POST"])
     def get_check():
-        signature = request.headers.get("Authorization")
-        password = str(base64.b64decode(signature), "utf-8")
-
-        if password != getenv("REQUEST_PASSWORD"):
-            return json.dumps({"success": False}), 403
+        if not validate_request(request.headers.get("Authorization")):
```
Comment (richardbm) [inline integration/app.py:106]: this service `validate_request` should raise the exception and you should catch it here and send to get_error_response the caught exception. maybe the exceptions for the satellite should have the status code to be returned. so you don't need to pass it, and you could be able to raise different exceptions that returns  different status codes

**Review by richardbm — CHANGES_REQUESTED** [review]

**Review by jeffersonlizar — APPROVED** [review]
LGTM

**Review by richardbm — APPROVED** [review]

**Pull request merged**